contracts/api3-server-v1/proxies/Api3CrossRateReaderProxyV1.sol

```diff
@@ -0,0 +1,195 @@
+    /// is not stale
+    /// @return value Cross rate between the two data feed values
+    /// @return timestamp Timestamp of the oldest data feed update
+    function read()
+        public
+        view
+        override
+        returns (int224 value, uint32 timestamp)
+    {
+        (int224 value1, uint32 timestamp1) = IApi3ReaderProxy(proxy1).read();
+        (int224 value2, uint32 timestamp2) = IApi3ReaderProxy(proxy2).read();
+
+        int256 val1 = int256(value1);
+        int256 val2 = int256(value2);
+        int256 crossRate;
+        if (calculationType == CalculationType.Divide1By2) {
+            if (val2 == 0) revert ProxyReturnedZero(proxy2);
+            crossRate = (val1 * 1e18) / val2;
+        } else if (calculationType == CalculationType.Divide2By1) {
+            if (val1 == 0) revert ProxyReturnedZero(proxy1);
+            crossRate = (val2 * 1e18) / val1;
+        } else if (calculationType == CalculationType.Multiply) {
+            crossRate = (val1 * val2) / 1e18;
+        }
+        if (crossRate > type(int224).max || crossRate < type(int224).min) {
+            revert ValueOutOfRange(crossRate);
+        }
+
+        value = int224(crossRate);
+        timestamp = timestamp1 < timestamp2 ? timestamp1 : timestamp2;
+    }
+
+    /// @dev AggregatorV2V3Interface users are already responsible with
+    /// validating the values that they receive (e.g., revert if the spot price
+    /// of an asset is negative). Therefore, this contract omits validation.
+    function latestAnswer() external view override returns (int256 value) {
+        (value, ) = read();
+    }
+
+    /// @dev A Chainlink feed contract returns the block timestamp at which the
+    /// feed was last updated. On the other hand, an API3 feed timestamp
+    /// denotes the point in time at which the first-party oracles signed the
+    /// data used to do the last update. We find this to be a reasonable
+    /// approximation, considering that usually the timestamp is only used to
+    /// check if the last update is stale.
+    function latestTimestamp()
+        external
+        view
+        override
+        returns (uint256 timestamp)
+    {
+        (, timestamp) = read();
+    }
+
+    /// @dev API3 feeds are updated asynchronously and not in rounds
+    function latestRound() external pure override returns (uint256) {
+        revert FunctionIsNotSupported();
+    }
+
+    /// @dev Functions that use the round ID as an argument are not supported
+    function getAnswer(uint256) external pure override returns (int256) {
+        revert FunctionIsNotSupported();
+    }
+
+    /// @dev Functions that use the round ID as an argument are not supported
+    function getTimestamp(uint256) external pure override returns (uint256) {
+        revert FunctionIsNotSupported();
+    }
+
+    /// @dev API3 feeds always use 18 decimals
+    function decimals() external pure override returns (uint8) {
+        return 18;
```

Comment:
I wonder if there are cases where the user requires different decimals

Reply:
I think that as long as proxies are API3 proxies then then answer is no

Reply:
I don't think any user contract was ever built anticipating to use Api3 proxies so they may have had to build an adapter for decimal conversion

Reply:
https://github.com/api3dao/contracts/pull/472/commits/012d779b03e408bde97ef75e23a558c698deab1d

Reply:
Related to https://github.com/api3dao/contracts/pull/472#discussion_r2070480945